pkgs/development/haskell-modules/configuration-common.nix

```
@@ -88,6 +88,13 @@ self: super: {
     };
   }) super.git-annex;
 
+  hlint = super.hlint.overrideAttrs(oa: {
+    postInstall = ''
+      mkdir -p $out/share/man
+      cp data/hlint.1 $out/share/man
+      '';
+  });
```

* `overrideCabal` makes sure modifications are not later rolled back by another chained haskell override (we want to override the args to `haskellPackages.mkDerivation`, not `stdenv.mkDerivation`).
* `drv` is used  generally as “old args” variable in the `configuration*` files, so stick with that.
* Make sure to concatenate the `postInstall` with the old one (if any).

```suggestion
  hlint = overrideCabal (drv: {
    postInstall = ''
      mkdir -p $out/share/man
      cp data/hlint.1 $out/share/man
    '' + drv.postInstall or "";
  }) super.hlint;
```

---

If you feel like it you could also use `install -Dm644 data/hlint.1 -t "$out/share/man"` to do directory creation and copying in one step.